litellm/llms/prompt_templates/factory.py

```diff
@@ -2346,6 +2374,10 @@ def _bedrock_tools_pt(tools: List) -> List[BedrockToolBlock]:
     for tool in tools:
         parameters = tool.get("function", {}).get("parameters", None)
         name = tool.get("function", {}).get("name", "")
+
+        # related issue: https://github.com/BerriAI/litellm/issues/5007
+        # Bedrock tool names must satisfy regular expression pattern: [a-zA-Z][a-zA-Z0-9_]* ensure this is true
+        name = make_valid_bedrock_tool_name(input_tool_name=name)
```

Comment:
@ishaan-jaff we need to output parse the tool call as well to make sure it's consistent with the user input 

<img width="1076" alt="Screenshot 2024-08-09 at 4 03 36 PM" src="https://github.com/user-attachments/assets/0d457bf5-c40d-46c3-a60e-7c640b76b4de">